workflows/sweep_simulate.snake

```diff
@@ -0,0 +1,432 @@
+        contig = species.get_contig(chrom, genetic_map=genetic_map_id, left=b_window_left, right=b_window_right)
+        if wildcards.dfe != "NA":
+            dfe = species.get_dfe(wildcards.dfe)
+
+            ## Adding annotation only seletion on exon region
+            annot = species.get_annotations(wildcards.annot)
+            annot_intervals = annot.get_chromosome_annotations(chrom)
+            contig.add_dfe(intervals=annot_intervals, DFE=dfe)
+
+        # sweep parameters
+        coord = ((window_left+window_right)//2) # the API does the right thing even if it is chunked
+        coeff = float(wildcards.coeff)
+        # Defining how long ago beneficial mutations arise
+        # First getting 2N from the model and pop
+        debug = model.model.debug()
+        coal_T = debug.mean_coalescence_time({wildcards.popu: 2})
+        gamma = coal_T* coeff #2Ns
+        T_f = 4 * (np.log(gamma) + 0.5772 - (1/gamma)) / coeff # this is from Charlesworth 2020/Pennings and Hermisson 2007
+        time = np.floor(1.5*T_f) # I dilated the time a bit because of variance (don't want to lose too many sims due to this -- note we are conditioning on sweep_min_freq)
+        print(coal_T, gamma, T_f, time)
+        contig.add_single_site("sweep", coordinate=coord)
+        extended_events = stdpopsim.ext.selective_sweep(
+            "sweep",
+            population=wildcards.popu, #selected pop
+            selection_coeff=coeff,
+            mutation_generation_ago=time,
+            min_freq_at_end=config["sweep_min_freq"]
+        )
+        
+        contig.mutation_rate = mutation_rate
+        engine = stdpopsim.get_engine("slim")
+        ts = engine.simulate(
+            model,
+            contig,
+            samples,
+            extended_events=extended_events,
+            slim_scaling_factor=config["slim_scaling_factor"],
+            slim_burn_in=config["slim_burn_in"],
+            seed=int(wildcards.seed),
+        )
+        ts = add_window_metadata(ts, window_left, window_right, b_window_left, b_window_right, extra=f"sweep_time={time};")
+        ts.dump(output[0])
+
+def _get_stats(input, output, params_dict, target_pops):
+    meta = ["model","demo","seed","chrom","left","right","annot","dfe","coeff"]
+    stats = ["taj"]
+    header = "\t".join(meta+['pop']+stats)
+    meta_str = "\t".join([params_dict[m] for m in meta])
+    ts = tskit.load(input[0])
+    all_pops_in_ts = [pop.metadata['name'] for pop in ts.populations()]
+    if target_pops is None:
+        target_pops = pop_list
+    sample_sets = [ts.samples(population=all_pops_in_ts.index(pop)) for pop in target_pops]
+    fh = open(output[0], 'w')
+    focal_taj, _ = compute_stats(ts, sample_sets)
+    assert focal_taj.shape == (len(target_pops),)
+    print(f'{header}', file=fh, flush=True)
+    for t,p in zip(focal_taj,target_pops):
+        print(f'{meta_str}\t{p}\t{t}', file=fh, flush=True)
+    fh.close()
+
+rule get_stats:
```

Comment:
in a subsequent PR I'm hoping to add a rule that would use `scikit-allel` to calculate stats like haplotype homozygosity. maybe change this rule name to `get_stats_tskit`?